### DISCUSSION TITLE
Can not import geometry remotely.

### DISCUSSION THREAD

**Discussion by narnia-ai-mason** [ansys/pymechanical#490] [Q&A]
Hi, I'm working on accessing Mechanical remotely.
I have two machines:
- `ansys_host`: Remote machine that Ansys has been installed.
- `client`: Local machine that I'm working on with Python and `ansys-mechanical-core` .

# Case I (directly executing `ansys-mechanical`)
First, I have run `ansys-mechanical --port <port>` directly on `ansys_host`.
Then `mech = launch_mechanical(ip=ansys_host, port=port)` works.
I have a IronPython script for importing geometry as follows:
```Python
GEO_IMPORT_SCRIPT = """
geo_import_format = Ansys.Mechanical.DataModel.Enums.GeometryImportPreference.Format.Automatic
geo_import_preferences = Ansys.ACT.Mechanical.Utilities.GeometryImportPreferences()
geo_import_preferences.ProcessSurfaces = False
geo_import_preferences.ProcessNamedSelections = True
geo_import_preferences.NamedSelectionKey = "" 
geo_import = Model.GeometryImportGroup.AddGeometryImport()
geo_import.Import(r"{cad_path:s}", geo_import_format, geo_import_preferences)
"""
```
So I can import a geometry by:
```Python
cad_path = "C:\\Users\\user\\workspace\\bracket0.scdoc"
mech.run_python_script(
    GEO_IMPORT_SCRIPT.format(cad_path=cad_path)
)
```
I've checked if the geometry was imported.
```Python
num_children = int(mech.run_python_script(
    'len(DataModel.Project.Model.Children)'
))
print('Model')
for i in range(num_children):
    name = mech.run_python_script(
        f'DataModel.Project.Model.Children[{i:d}].Name'
    )
    print(f'|- {name}')
    num_subchildren = int(mech.run_python_script(
        f'len(DataModel.Project.Model.Children[{i:d}].Children)'
    ))
    for j in range(num_subchildren):
        child_name = mech.run_python_script(
            f'DataModel.Project.Model.Children[{i:d}].Children[{j:d}].Name'
        )
        print(f'|  |- {child_name}')
```
It gave me
```
Model
|- Geometry Imports
|  |- Geometry Import
|- Geometry
|  |- component
|- Materials
|  |- Structural Steel
|- Coordinate Systems
|  |- Global Coordinate System
|- Remote Points
|- Mesh
|- Named Selections
|  |- Remained
|  |- Fixed
|  |- LoadedL
|  |- LoadedR
|  |- Loaded
```
It means the geometry has been successfully imported!
# Case II (remotely executing `ansys-mechanical` using SSH)
Secend, I have run `ansys-mechanical --port <port>` remotely on `ansys_host` using **SSH**, and did the same thing.
But I got
```
Model
|- Geometry Imports
|  |- Geometry Import
|- Geometry
|- Materials
|- Coordinate Systems
|  |- Global Coordinate System
|- Remote Points
|- Mesh
```
It means, the `Geometry Import` has been added (`geo_import = Model.GeometryImportGroup.AddGeometryImport()`),
but the importing itself has failed (`geo_import.Import(r"{cad_path:s}", geo_import_format, geo_import_preferences)`).

Why this happen and how can I resolve this problem?

**Comment by dipinknair** [comment]
@narnia-mason Hi, What Operating Systems are you using for ``ansys_host`` and ``client``?
can you provide any logs?

**Reply by narnia-ai-mason** [reply]
Hi! `ansys_host` is Windows machine and I tested on Mac, Windows, and Ubuntu for the `client`.

**Reply by dipinknair** [reply]
@narnia-mason can you please send an email to pyansys.core@ansys.com so that we can communicate over email directly. I have connected to Linux machine host through ssh and tried your code, imported a different geometry and did not see any issue.

**Comment by RobPasMue** [comment]
Hi @narnia-mason - just guessing here, but is the ``cad_path`` file present in your remote machine as well, with the same path? If the file does not exist on the remote machine, I don't think the remote execution mechanism for ``ansys-mechanical-core`` would also figure out that it needs to send the CAD file to the remote machine.
